File: python/task4.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        j = 1
        c = 1

        for i in range(1, len(nums)):
            if (nums[i] == nums[i-1]) and c == 1:
                nums[j] = nums[i]
                c += 1
                j += 1
            elif (nums[i] != nums[i-1]):
                nums[j] = nums[i]
                j += 1
                c = 1
        return j

File: c++/algorithms/task4.cpp
#include <vector>
class Solution {
public:
    int removeDuplicates(std::vector<int>& nums) {
        int j = 1;
        int c = 1;
        for (int i = 1; i < nums.size(); i++){
            if ((nums[i] == nums[i-1]) && c == 1){
                nums[j] = nums[i];
                j++;
                c++;
            }
            else if (nums[i] != nums[i-1]){
                nums[j] = nums[i];
                c = 1;
                j++;
            }
        }
        return j;
    }
};